# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/gcc3/protector-docs.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./gcc/doc/gcc.1.orig	2002-11-19 19:17:19.000000000 +0100
+++ ./gcc/doc/gcc.1	2003-01-02 16:08:53.000000000 +0100
@@ -687,8 +687,8 @@
 \&\-fno-common  \-fno-ident  \-fno-gnu-linker 
 \&\-fpcc-struct-return  \-fpic  \-fPIC 
 \&\-freg-struct-return  \-fshared-data  \-fshort-enums 
-\&\-fshort-double  \-fshort-wchar \-fvolatile 
-\&\-fvolatile-global  \-fvolatile-static 
+\&\-fshort-double  \-fstack-protector  \-fshort-wchar
+\&\-fvolatile  \-fvolatile-global  \-fvolatile-static 
 \&\-fverbose-asm  \-fpack-struct  \-fstack-check 
 \&\-fstack-limit-register=\fR\fIreg\fR  \fB\-fstack-limit-symbol=\fR\fIsym\fR 
 \&\fB\-fargument-alias  \-fargument-noalias 
@@ -8954,6 +8954,12 @@
 instructions to throw exceptions, i.e. memory references or floating
 point instructions.  It does not allow exceptions to be thrown from
 arbitrary signal handlers such as \f(CW\*(C`SIGALRM\*(C'\fR.
+.Ip "\fB\-fstack-protector\fR" 4
+.IX Item "-fstack-protector"
+Enable the Stack-Smashing Protector (
+.B http://www.trl.ibm.com/projects/security/ssp/
+). This option is only available in ROCK Linux (and probably
+some other distributions). It's not a standard GCC option.
 .Ip "\fB\-funwind-tables\fR" 4
 .IX Item "-funwind-tables"
 Similar to \fB\-fexceptions\fR, except that it will just generate any needed
--- ./gcc/doc/invoke.texi.orig	2002-10-29 17:35:04.000000000 +0100
+++ ./gcc/doc/invoke.texi	2003-01-02 16:12:11.000000000 +0100
@@ -676,7 +676,7 @@
 -fno-common  -fno-ident  -fno-gnu-linker @gol
 -fpcc-struct-return  -fpic  -fPIC @gol
 -freg-struct-return  -fshared-data  -fshort-enums @gol
--fshort-double  -fshort-wchar -fvolatile @gol
+-fshort-double  -fstack-protector  -fshort-wchar -fvolatile @gol
 -fvolatile-global  -fvolatile-static @gol
 -fverbose-asm  -fpack-struct  -fstack-check @gol
 -fstack-limit-register=@var{reg}  -fstack-limit-symbol=@var{sym} @gol
@@ -9721,6 +9721,14 @@
 it.
 
 @table @gcctabopt
+
+@item -fstack-protector
+@opindex fstack-protector
+Enable the Stack-Smashing Protector
+(http://www.trl.ibm.com/projects/security/ssp/).
+This option is only available in ROCK Linux (and probably
+some other distributions). It's not a standard GCC option.
+
 @item -fexceptions
 @opindex fexceptions
 Enable exception handling.  Generates extra code needed to propagate
